feat(local-install): Add `npmEnv` option to LocalInstaller

wojtkowiak opened the pull request:
Hello,

I have a plan to use your package in `meteor-desktop` but I've faced a problem when installing local packages that have native node modules. The compilation goes against current node and what I need is to compile them against the electron version. For that I need to pass additional env vars to npm. I have made this small change to be able to do that.
I am a total newbie in typescript so I hope its typescriptish enough 🙂
Let me know if that can go into your next release.

nicojs commented on src/LocalInstaller.ts:
```diff
@@ -11,17 +12,31 @@ interface PackageByDirectory {
+    constructor(sourcesByTarget: ListByPackage, options?: Options) {
         super();
         this.sourcesByTarget = resolve(sourcesByTarget);
+        if (options) {
```
Could we use Object.assign here? `options = Object.assign({}, options);`

wojtkowiak replied:
sure, good point

nicojs replied:
`Object.assign` can handle an `undefined` value. So you can even remove this entire `if`:

```ts
 constructor(sourcesByTarget: ListByPackage, options?: Options) {
         super();
         this.sourcesByTarget = resolve(sourcesByTarget);
         this.options = Object.assign({}, options);
     }
```

nicojs commented on src/LocalInstaller.ts:
```diff
@@ -1,3 +1,4 @@
+import * as child_process from 'child_process';
```
the `ExecOptions` are actually re-exported from `mz/child_process` (using `export * from "child_process";`), so you should be able to use `import { exec, ExecOptions } from 'mz/child_process';` instead.

wojtkowiak replied:
Oww, good to know, much more cleaner that way.

nicojs commented:
You've got to love mutation testing. Build failed because new code paths do not have unit tests:

```
Final mutation score 71.43 under breaking threshold 72, setting exit code to 1 (failure).
```

wojtkowiak commented:
Thanks for looking into this 🙂 
Sorry for the delay, I needed to handle some other issues.
I have adjusted the commit according to your comments and added the test. I have also added a section in the Readme. Feel free to modify it.

> Are you planning to use it from the programmatic API and not the CLI?

I guess it would be difficult to use it from the CLI. I mean passing an object from CLI would be hard. Maybe there should be an option to pass it when using the CLI interface programmatically but not sure if it's worth it.

nicojs commented:
It's released in v0.5.0: https://github.com/nicojs/node-install-local/blob/master/CHANGELOG.md

wojtkowiak commented:
Thanks for your time and for merging it in! 👍
